src/networking.rs

```diff
@@ -0,0 +1,54 @@
+use reqwest::{blocking::{Client, Response}, Error};
+
+static BASE_URL: &str = "https://18ye1iivg6.execute-api.us-west-1.amazonaws.com/v4";
+
+fn get_reports(interval_number: u32, interval_length: u32) -> Result<Vec<String>, Error> {
+  let url: &str = &format!("{}/tcnreport", BASE_URL);
+
+  create_client().and_then(|client| 
+      client.get(url)
+          .header("Content-Type", "application/json")
+          .query(&[("intervalNumber", interval_number)])
+          // TODO: Will be changed to seconds
+          .query(&[("intervalLengthMs", interval_length)])
+          .send()
+          .and_then (|response| response.json::<Vec<String>>())
+  )
+}
+
+fn post_report(report: &'static str) -> Result<Response, Error> {
+  let url: &str = &format!("{}/tcnreport", BASE_URL);
+
+  create_client().and_then(|client| 
+      client.post(url)
+          .header("Content-Type", "application/json")
+          .body(report)
+          .send()
+  )
+}
+
+fn create_client() -> Result<Client, Error> {
+  reqwest::blocking::Client::builder()
+    // .proxy(reqwest::Proxy::https("http://localhost:8888")?) // Charles proxy
+    .build()
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use reqwest::StatusCode;
+
+  #[test]
+  fn get_reports_is_ok() {
+    let res = get_reports(1, 21600000);
```

Comment:
@i-schuetz Can we log/print request and response?

Reply:
I assume ;)